tactics/class_tactics.ml

```diff
@@ -932,11 +932,12 @@ module Search = struct
        try
          (* Instance may try to call this before a proof is set up!
             Thus, give_me_the_proof will fail. Beware! *)
-         let name, poly = try
-             let Proof.{ name; poly } = Proof.data Proof_global.(give_me_the_proof ()) in
-             name, poly
-           with | Proof_global.NoCurrentProof ->
-             Id.of_string "instance", false
+         let name, poly =
+           (* try
+            *   let Proof.{ name; poly } = Proof.data Proof_global.(give_me_the_proof ()) in
+            *   name, poly
+            * with | Proof_global.NoCurrentProof -> *)
+           Id.of_string "instance", false
```

Comment:
change of semantics, in particular `poly` is now always false

Reply:
Indeed that means we cannot support `abstract` in polymorphic mode when happening as part of the pretyper call. I dunno if that even works today as it doesn't seem to be tested.

The fix here is far from trivial, as it was outlined in the comment that would imply a rework of the tactics hook into the pretyper.

Reply:
... or rewrite the pretyper in the monad...

Reply:
Well, that's a big hammer for the current problem which has to do more with the current implementation of abstract being problematic IMO, but yeah, if you can do that without getting 50% slower that'd be nice, even tho I am very wary of the current implementation of the monad, in particular its lack of support for dependent refinement.

But indeed I had forgotten that I've added a patch so the pretyper knows about polymorphic mode [needed for `ltac:` to work], so indeed maybe this can be fixed more easily.

For sure, the current hook has to be fixed regardless of whether the pretyper goes to the monad or not.

Reply:
So, what should be done about this?

Reply:
@ppedrot or @mattam82 can you confirm this change is ok with you? There does not seem to be an easy way to do better.